addons/mail/models/mail_link_preview.py

```diff
@@ -0,0 +1,152 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from lxml import html, etree
+from urllib.parse import urlparse
+import requests
+
+from odoo import api, models, fields
+
+
+class LinkPreview(models.Model):
+    _name = 'mail.link.preview'
+    _description = "Store link preview data"
+
+    message_id = fields.Many2one('mail.message', string='Message', index=True, ondelete='cascade', required=True)
+    source_url = fields.Char('url', required=True)
+    og_type = fields.Char('type')
+    og_title = fields.Char('title')
+    og_image = fields.Char('image')
+    og_description = fields.Text('description')
+    og_mimetype = fields.Char('mimetype')
+    image_mimetype = fields.Char('image_mimetype')
+    create_date = fields.Datetime(index=True)
+
+    requestsSession = requests.Session()
```

Comment:
@xmo-odoo did you imply the session should be global like this, or to have one session per link preview request loop?

Reply:
lint: snake case

Reply:
> did you imply the session should be global like this, or to have one session per link preview request loop?

The latter. IIRC `requests.Session` is *not* thread-safe, and furthermore setting it as a class attribute opens up security issues.

Reply:
I have made the change, but I m not too happy about having to passe `request_session` as parameter of half of the methods.